DKBeatMaker/DKBeatMaker/ArrangementView/ArrangementViewController.swift

```diff
@@ -0,0 +1,61 @@
+import UIKit
+
+let midiClipReuseIdentifier: String = "MidiClipCell"
+let audioControllerReuseIdentifier: String = "audioControllerCell"
+
+class ArrangementViewController: UICollectionViewController, AddClipDelegate, ArrangementViewPresenterDelegate{
+    var arrangementViewPresenter: ArrangementViewPresenter?
+    
+    @IBOutlet weak var multiDirectionalLayout: MultiDirectionalCollectionViewLayout!
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.arrangementViewPresenter = ArrangementViewPresenter.init()
+        self.arrangementViewPresenter?.delegate = self
+        
+    }
+    
+    // MARK: UICollectionViewDataSource
+    override func numberOfSectionsInCollectionView(collectionView: UICollectionView) -> Int {
+        return (self.arrangementViewPresenter?.sections.count)!
+    }
+    
+    
+    override func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return (self.arrangementViewPresenter?.numberOfItemForSection(section))!
+    }
+    
+    override func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath) -> UICollectionViewCell {
+        if indexPath.row == 0{
+            let cell = collectionView.dequeueReusableCellWithReuseIdentifier(audioControllerReuseIdentifier, forIndexPath: indexPath) as! AudioControllerCell
+            cell.trackNumber = indexPath.section
+            cell.delegate = self
+            return cell
+        }else{
+            let cell = collectionView.dequeueReusableCellWithReuseIdentifier(midiClipReuseIdentifier, forIndexPath: indexPath) as! MidiClipCollectionViewCell
+            cell.label.text = "Sec \(indexPath.section)/Item \(indexPath.item)"
+            return cell
+        }
+    }
+    
+    // MARK: UICollectionViewDelegate
+    
+    override func collectionView(collectionView: UICollectionView, didSelectItemAtIndexPath indexPath: NSIndexPath) {
+        if(indexPath.row != 0){
```

Comment:
Instead of determining what happens when a UICollectionView item is selected, I would just notify the presenter that a particular item was clicked and let the presenter decide where to segue to. 
When we had the MVP discussion, Steve thought this was excessive. However, thats how I would do it because any UI event IMO should be handled by the presenter. The view controller has no decision-making capacity.